apps/website/src/composables/wallet.ts

```diff
@@ -1,47 +1,82 @@
 import { ref } from 'vue'
 import { ethers } from 'ethers'
-import { WalletProvider } from '@/interfaces/WalletProvider'
+import useIopay from '@/composables/iopay'
+import useEthers from '@/composables/ethers'
+
+const { requestEthersAccount } = useEthers()
+const { getIoPayAccounts, sendIoPayTransaction } = useIopay()
 
 const defaultProviders = {
-  metamask: undefined,
-  coinbase: undefined,
+  MetaMask: undefined,
+  CoinbaseWallet: undefined,
 }
 
 export default function useWallet() {
   const ethereum: any = window.ethereum
-  const availableProviders = ref<Record<string, WalletProvider>>(
-    getAvailableProviders(ethereum)
-  )
-  const selectedProvider = ref<WalletProvider>({})
+  const availableProviders = ref<any>(getAvailableProviders(ethereum)) // TODO: Retype this?
+  const selectedProvider = ref<'MetaMask' | 'CoinbaseWallet' | 'IoPay' | ''>('')
   const selectedAccount = ref<string>('')
-  const toAddress = ref<string>('') // Test to address: 0xD4e5faa8aD7d499Aa03BDDE2a3116E66bc8F8203
   const amount = ref<string>('')
+  const toAddress = ref<string>('')
+  // Test ethereum send to address : 0xD4e5faa8aD7d499Aa03BDDE2a3116E66bc8F8203
+  // Test iotex send to address: acc://06da5e904240736b1e21ca6dbbd5f619860803af04ff3d54/acme
 
   async function connectWallet(provider: string) {
     try {
-      selectedProvider.value = availableProviders.value[provider]
-      if (!selectedProvider.value) {
+      if (provider === 'MetaMask') {
+        selectedProvider.value = 'MetaMask'
+        const browserExtensionProvider = availableProviders.value[provider]
+        selectedAccount.value = (
```

Comment:
Create a `setSelectedAccount` that accepts a valid address.

Reply:
Move account request before `setSelectedAccount`. 

You'll need to assert the browser provider type `const account = await requestEthersAccount(browserProvider as BrowserProvider)`.